src/fides/api/alembic/migrations/versions/14acee6f5459_translation_data.py

```diff
@@ -0,0 +1,651 @@
+        _raw_experience_config_map[experience_config_type] = experience_config.copy()
+
+        # the reconciled experience config will have its regions populated by existing notices that are
+        # associated with this experience config. so in most cases, we start "fresh" with an empty set.
+        # the TCF experience is an exception - it is not linked to any notices, and therefore its
+        # regions are not derived from notices, so we hardcode its regions to the config default.
+        reconciled_regions = (
+            experience_config["regions"]
+            if experience_config_type == DefaultExperienceConfigTypes.EEA_TCF_OVERLAY
+            else set()  # if not TCF experience, clear the config's regions, since we will populate based on existing notices
+        )
+        experience_config["regions"] = reconciled_regions
+        experience_config["privacy_notices"] = set()  # initialize the notices field
+        experience_config["needs_migration"] = (
+            False  # indicator whether the record requires migration, or we can default to the template values
+        )
+        _reconciled_experience_config_map[experience_config_type] = experience_config
+        experience_config["needs_migration"] = False
+    return _reconciled_experience_config_map, _raw_experience_config_map
+
+
+reconciled_experience_config_map, raw_experience_config_map = (
+    load_default_experience_configs()
+)
+
+
+def backup_existing_data_csv(bind):
+    """
+    Perform a backup (dump to csv) of the data in our tables that will be impacted by migration.
+
+    Because this data migration is risky, it is recommended to perform a comprehensive DB backup/snapshot beforehand!
+    The "automated" dump to CSV here is an additional safety measure.
+    """
+    for table_name in [
+        "privacyexperience",
+        "privacyexperienceconfig",
+        "privacyexperienceconfighistory",
+        "privacynotice",
+        "privacynoticehistory",
+    ]:
+        dump_table_to_csv(bind, table_name)
+
+
+def remove_existing_experience_data(bind):
+    """
+    We remove any existing PrivacyExperience and PrivacyExperienceConfig records,
+    to allow for a "clean-slate" in creation of _new_ PrivacyExperienceConfigs based on the
+    "out of the box" templates.
+
+    This means that any existing user-edited data on PrivacyExperienceConfig records will need to be
+    _manually_ re-added to the post-migration state. It will not be migrated as part of the automatic migration.
+    """
+
+    def delete_experiences(bind):
+        """
+        Delete all PrivacyExperience records.
+
+        These records will be effectively re-created as a result of the "out of the box" creation
+        of 'new-style' PrivacyExperienceConfig records post-migration.
+        """
+        bind.execute(
+            text(
+                """
+                DELETE FROM privacyexperience;
```

Comment:
The previous schema migration drops the FK constraint on PrivacyPreferenceHistory.privacy_experience because we're not necessarily persisting those resources anymore, and we don't really need it with the combo of the user geography and the privacy experience config history id.  

That also allowed this action to be possible

I think we should probably delete this column now in the followup? Are there slowness risks though both for removing the FK and/or deleting a column from a large table?

Reply:
EDIT: OK my third follow-up migration now deletes this obsolete column from PrivacyPreferenceHistory and ServedNoticeHistory https://github.com/ethyca/fides/pull/4651/files.

We should still double check if the migration to remove the FK and/or delete the column is okay on these big tables - I think deleting the FK could lock the table. Part of the migration plan could be taking some downtime so this table is not being concurrently accessed.

Reply:
ok got it, either of those approaches seems viable to me - allotting for some downtime for the migration may be prudent regardless of this particular point. let's do some more exploration and plan to finalize 👍